### PR TITLE
fix custom shader imports

### PR DESCRIPTION
**Pull request by robtfm**
# Objective

assets v2 broke custom shader imports. fix them

## Solution

store handles of any file dependencies in the `Shader` to avoid them being immediately dropped.
also added a use into the `shader_material` example so that it'll be harder to break support in future.

### CODE REVIEW

File: crates/bevy_render/src/render_resource/shader.rs [hunk 1]
```diff
@@ -270,11 +276,10 @@ impl AssetLoader for ShaderLoader {
                 _ => panic!("unhandled extension: {ext}"),
             };
 
-            // collect file dependencies
+            // collect and store file dependencies
             for import in &shader.imports {
                 if let ShaderImport::AssetPath(asset_path) = import {
-                    // TODO: should we just allow this handle to be dropped?
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/render_resource/shader.rs:276]: Very funny TODO.

File: assets/shaders/custom_material_import.wgsl [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// this is made available to the importing module
+const MULT: vec4<f32> = vec4<f32>(1.0, 1.0, 1.0, 0.5);
```
Comment (alice-i-cecile) [inline assets/shaders/custom_material_import.wgsl:2]: This is a weird and unclear variable name. Can we change it to something more semantic?

**Review by alice-i-cecile — APPROVED** [review]
Fix makes sense to me. Nothing blocking, but improving the variable name would be nice.

File: assets/shaders/custom_material_import.wgsl (hunk 2 above)
Comment (robtfm) [reply assets/shaders/custom_material_import.wgsl:2]: sure, how about `COLOR_MULTIPLIER` or `EXAMPLE_IMPORTED_COLOR_MULTIPLIER` ?

Comment (alice-i-cecile) [reply assets/shaders/custom_material_import.wgsl:2]: Both are fine. The former is probably adequate.

File: assets/shaders/custom_material_import.wgsl [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// this is made available to the importing module
+const COLOR_MULTIPLIER: vec4<f32> = vec4<f32>(1.0, 1.0, 1.0, 0.5);
```
Comment (superdump) [inline assets/shaders/custom_material_import.wgsl:2]: Missing newline. Maybe configure your editor to ensure there is a newline at the end of files? :)

**Review by superdump — CHANGES_REQUESTED** [review]
Very minor nit, otherwise approved. Thanks for putting this together!

File: assets/shaders/custom_material_import.wgsl (hunk 3 above)
Comment (robtfm) [reply assets/shaders/custom_material_import.wgsl]: done and done.

**Review by superdump — APPROVED** [review]

**Pull request merged**